collectors/CMPCollector.js

```diff
@@ -344,14 +341,14 @@ class CMPCollector extends BaseCollector {
                 const matches = allTexts.match(p);
                 if (matches) {
                     foundPatterns.push(p.toString());
-                    foundSnippets.push(...matches);
+                    foundSnippets.push(...matches.map(m => m.substring(0, 200)));
                 }
             }
         }
         this.pendingScan.resolve();
-        return {
+        this.scanResult = {
             patterns: foundPatterns,
-            snippets: foundSnippets,
+            snippets: Array.from(new Set(foundSnippets)),
```

Comment:
one bit of info is lost in thie process, now we won't have 1:1 relationship between patterns and snippets (patterns[0] <=> snippets[0], …), but probably that's OK?